### ISSUE TITLE
Knex migrate:* attempts to load '.d.ts' files

### ISSUE THREAD

**Issue by Kematia** [knex/knex#4517]
# Environment

Knex version: 0.95.6
Database + version: 
OS: Ubuntu via WSL2 (Win 10)

Select applicable template from below.
If issue is about oracledb support, tag @ atiertant. For MSSql tag @ smorey2.
Rest of dialects don't need tags.
If issue is about TypeScript definitions, tag @lorefnon .

Reproducible repo: https://github.com/Kematia/knex-loads-types

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

`knex migrate:*` attempts to run a `.d.ts` file if it's present in the `migrations` folder

2. Error message

![image](https://user-images.githubusercontent.com/17851386/121308668-9660ab00-c901-11eb-8ffe-3e74dbf497b6.png)
![image](https://user-images.githubusercontent.com/17851386/121308687-9bbdf580-c901-11eb-82eb-d62a5fd4c925.png)

If I manually delete the `.d.ts` file it works correctly:

![image](https://user-images.githubusercontent.com/17851386/121308874-d58efc00-c901-11eb-9572-6c885b5c24d6.png)
![image](https://user-images.githubusercontent.com/17851386/121308893-daec4680-c901-11eb-9ad7-ba6a41fa4d03.png)

**Comment by Mahdhir** [comment]
I can look into this.

Any guidance on where I should get started will be appreciated.

**Comment by Kematia** [comment]
> I can look into this.
> 
> Any guidance on where I should get started will be appreciated.

I've set up a repo to reproduce the error: https://github.com/Kematia/knex-loads-types

**Comment by Mahdhir** [comment]
@Kematia 
I added `loadExtensions:['.js']` to knexfile.js and your repo worked without producing an error.

```
migrations: {
          tableName: "knex_migrations",
          directory: "./dist/migrations",
          loadExtensions:['.js']
      },
```

And I had to install `ts-node` as well.
Not exactly sure if this is a bug or expected behavior

**Comment by Kematia** [comment]
`loadExtensions:['.js']` seems like will fix my issue! Thank you :)

I do think that knex trying to load `.d.ts` files is a bug, it should try to load either `.ts` or `.js` files.

**Comment by elhigu** [comment]
Yep... this issue actually yells for support to give regexp style matching to allow more complex load patterns. 

Current config doesn't allow to distinguish for example  `.generated.js` and `.d.ts` from desired `.js` and `.ts` files

**Comment by mercmobily** [comment]
Closing this in favour of #2756 which covers the same issue
